libqtile/lazy.py

```diff
@@ -18,7 +18,19 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-from typing import Dict, Iterable, List, Optional, Set, Tuple, Union
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+)
+
+if TYPE_CHECKING:
+    from libqtile.config import Match
```

Comment:
While you're here, could you put all of the other `typing` imports within the `if TYPE_CHECKING` block?

Reply:
Ah yeah will do, I needed it for Match to prevent the circular dependency

Reply:
Implemented.